pkg/ss/partition/fa.go

```diff
@@ -18,9 +18,13 @@ func (fa *followingActor) become() error {
 			pa := &primaryActor{fa.actorBase}
 			pa.setState(Primary)
 			return pa.become()
-		case common.ClientMsg:
-			// we needn't handle this because primary would've updated the state.
-			// todo what should we do when primary is in different node?
+		case *common.ProxiedMsg:
+			// todo it looks a bit odd for follower to process client messages, but this is ensuring
+			// the correctness of algorithm. we can revisit this.
+			pm := m.(*common.ProxiedMsg)
```

Comment:
you can use `switch v := m.(type)` and use `v` instead of typecasting again. neater and less error prone

Reply:
we won't be able to cast a struct to v in this case.